pandas/_libs/tslibs/offsets.pyx

```diff
@@ -313,6 +313,37 @@ class _BaseOffset(object):
     def __setattr__(self, name, value):
         raise AttributeError("DateOffset objects are immutable.")
 
+    def __eq__(self, other):
+        if is_string_object(other):
+            from pandas.tseries.frequencies import to_offset
+            other = to_offset(other)
+
+        try:
+            return self._params == other._params
+        except AttributeError:
+            # other is not a DateOffset object
+            return False
+
+        return self._params == other._params
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __hash__(self):
+        return hash(self._params)
+
+    @property  # NB: non-cython subclasses override with cache_readonly
+    def _params(self):
+        all_paras = self.__dict__.copy()
```

Comment:
can you give an expl of what this is doing (and you can move the comment inside the function too)